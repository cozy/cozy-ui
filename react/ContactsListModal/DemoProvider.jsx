import React from 'react'
import contacts from '../ContactsList/_mockContacts.json'
import CozyClient, { CozyProvider } from 'cozy-client'

const mockClient = new CozyClient({
  uri: 'http://cozy.tools:8080',
  token: 'faketoken'
})

mockClient.__proto__.requestQuery = ({ doctype }) => {
  if (doctype === 'io.cozy.contacts') {
    return Promise.resolve({
      fetchStatus: 'loaded',
      data: contacts
    })
  }

  if (doctype === 'io.cozy.apps') {
    return Promise.resolve({
      fetchStatus: 'loaded',
      data: [
        {
          _id: 'io.cozy.apps/contacts',
          _type: 'io.cozy.apps',
          attributes: {
            slug: 'contacts'
          },
          links: {
            related: 'http://contacts.cozy.tools:8080/'
          }
        }
      ]
    })
  }
}

mockClient.plugins = {
  realtime: {
    subscribe: () => {},
    unsubscribe: () => {},
    unsubscribeAll: () => {}
  }
}

mockClient.save = () => ({ data: contacts[0] })

const Wrapper = ({ children }) => {
  return <CozyProvider client={mockClient}>{children}</CozyProvider>
}

export default Wrapper
