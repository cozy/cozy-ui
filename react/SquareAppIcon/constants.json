{
  "iconSize": "4rem",
  "iconPadding": "0.625rem",
  "mobileIconSize": "3rem",
  "mobileIconPadding": "0.5rem",
  "color": "#fff"
}
