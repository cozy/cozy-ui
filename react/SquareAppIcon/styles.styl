@require 'settings/breakpoints.styl'
constants=json('constants.json', { hash: true })

$iconSize = constants['iconSize']
$iconPadding = constants['iconPadding']
$mobileIconSize = constants['mobileIconSize']
$mobileIconPadding = constants['mobileIconPadding']
$color = constants['color'] // hard-coded color, because the component is currently only used in the Home

.SquareAppIcon-wrapper
    box-sizing border-box
    background-color var(--paperBackgroundColor)
    border-radius rem(12)
    height $iconSize
    width $iconSize
    +small-screen()
        height $mobileIconSize
        width $mobileIconSize

.SquareAppIcon-icon-container
    display flex
    align-items center
    justify-content center
    padding $iconPadding
    border-radius rem(12)
    width 100%
    +small-screen()
        padding $mobileIconPadding
    svg, img
        width 100%

.SquareAppIcon-wrapper-fx
    background-color alpha($color, .32)
    border 1px dashed var(--white)
    background-color alpha($color, .48)

.SquareAppIcon-wrapper-ghost
    .SquareAppIcon-icon-container
        mix-blend-mode screen
        svg, img
            filter saturate(0%)

.SquareAppIcon-wrapper-maintenance
    .SquareAppIcon-icon-container
        mix-blend-mode luminosity
        svg, img
            opacity .5
