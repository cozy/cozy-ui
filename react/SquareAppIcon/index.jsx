import React from 'react'
import cx from 'classnames'
import PropTypes from 'prop-types'
import get from 'lodash/get'

import { makeStyles } from '../styles'
import AppIcon from '../AppIcon'
import Badge from '../Badge'
import InfosBadge from '../InfosBadge'
import { nameToColor } from '../Avatar'
import Spinner from '../Spinner'
import Typography from '../Typography'
import Icon from '../Icon'
import iconPlus from '../Icons/Plus'
import iconWarning from '../Icons/WarningCircle'
import iconOut from '../Icons/LinkOut'

import { color } from './constants.json'
import styles from './styles.styl'

const useStyles = makeStyles(theme => ({
  name: {
    color,
    width: '5.5rem',
    textAlign: 'center',
    fontSize: '0.875rem',
    lineHeight: '1.188rem',
    margin: '0.5rem 0.25rem 0 0.25rem',
    textShadow: theme.textShadows[1],
    lineClamp: '2',
    boxOrient: 'vertical',
    display: '-webkit-box',
    height: '2.375rem',
    [theme.breakpoints.down('sm')]: {
      width: '3.75rem',
      fontSize: '0.6875rem',
      lineHeight: '1rem',
      margin: '0.25rem 0.25rem 0 0.25rem',
      height: '2rem'
    }
  },
  letter: {
    color,
    margin: 'auto'
  },
  shadow: {
    boxShadow: theme.shadows[2]
  },
  errorIcon: {
    fill: 'var(--errorColor)',
    backgroundColor: 'var(--primaryContrastTextColor)',
    borderRadius: '1rem'
  },
  tileWrapper: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    width: '6rem',
    [theme.breakpoints.down('sm')]: {
      width: '4.25rem'
    }
  }
}))

export const SquareAppIcon = ({
  name,
  variant,
  IconContent,
  ...appIconProps
}) => {
  const classes = useStyles()
  const appName =
    name || get(appIconProps, 'app.name') || get(appIconProps, 'app') || ''
  const letter = appName[0] || ''

  return (
    <div data-testid="square-app-icon" className={cx(classes.tileWrapper)}>
      <InfosBadge
        badgeContent={
          variant === 'shortcut' ? <Icon size="10" icon={iconOut} /> : null
        }
        overlap="rectangular"
        invisible={variant !== 'shortcut'}
      >
        {['default', 'loading'].includes(variant) && (
          <Spinner className={cx(styles['SquareAppIcon-spinner'], 'u-m-0')} />
        )}
        <Badge
          className={cx(
            styles['SquareAppIcon-wrapper'],
            styles[`SquareAppIcon-wrapper-${variant}`],
            {
              [`${styles['SquareAppIcon-wrapper-fx']}`]: [
                'ghost',
                'add'
              ].includes(variant),
              [classes.shadow]: !['add', 'ghost'].includes(variant)
            }
          )}
          badgeContent={
            variant === 'error' ? (
              <Icon
                size="16"
                className={cx(classes.errorIcon)}
                icon={iconWarning}
              />
            ) : (
              ''
            )
          }
          color={variant === 'error' ? 'error' : undefined}
          withBorder={false}
          size="large"
          overlap="rectangular"
          style={
            variant === 'shortcut' && !IconContent
              ? { backgroundColor: nameToColor(name) }
              : null
          }
        >
          {variant === 'shortcut' && !IconContent ? (
            <Typography className={classes.letter} variant="h2" align="center">
              {letter.toUpperCase()}
            </Typography>
          ) : (
            <div className={styles['SquareAppIcon-icon-container']}>
              {variant === 'add' ? (
                <Icon icon={iconPlus} color={color} />
              ) : IconContent ? (
                IconContent
              ) : (
                <AppIcon {...appIconProps} />
              )}
            </div>
          )}
        </Badge>
      </InfosBadge>
      <Typography
        className={cx(classes.name, 'u-spacellipsis')}
        variant="h6"
        align="center"
      >
        {appName}
      </Typography>
    </div>
  )
}

SquareAppIcon.propTypes = {
  name: PropTypes.string,
  variant: PropTypes.oneOf([
    'default',
    'loading',
    'ghost',
    'maintenance',
    'error',
    'add',
    'shortcut'
  ]),
  IconContent: PropTypes.node
}

SquareAppIcon.defaultProps = {
  variant: 'default'
}

export default SquareAppIcon
