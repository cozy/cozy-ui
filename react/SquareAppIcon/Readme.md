Component used to show an app status, can be used in a list of apps and connectors in the Home
application.

```jsx
import Grid from 'cozy-ui/transpiled/react/MuiCozyTheme/Grid'
import SquareAppIcon from 'cozy-ui/transpiled/react/SquareAppIcon'
import CozyIcon from 'cozy-ui/transpiled/react/Icons/Cozy'
import Icon from 'cozy-ui/transpiled/react/Icon'
import { useCozyTheme } from 'cozy-ui/transpiled/react/CozyTheme'
import cloudWallpaper from '../../docs/cloud-wallpaper.jpg'

const theme = useCozyTheme()
const app = { name: "Test App", slug: "testapp", type: "app" }

;


<Grid container spacing={1} style={{ background: `center / cover no-repeat url(${cloudWallpaper})` }}
>
  <Grid item>
    <SquareAppIcon app={app} name="Normal" />
  </Grid>
  <Grid item>
    <SquareAppIcon app={app} name="Maintenance" variant="maintenance" />
  </Grid>
  <Grid item>
    <SquareAppIcon app={app} name="Error" variant="error" />
  </Grid>
  <Grid item>
    <SquareAppIcon name="Add" variant="add" />
  </Grid>
  <Grid item>
    <SquareAppIcon app="testapp" name="No Account long name very very very very long" variant="ghost" />
  </Grid>
  <Grid item>
    <SquareAppIcon name="Shortcut" variant="shortcut" />
  </Grid>
  <Grid item>
    <SquareAppIcon name="Custom Icon" IconContent={<Icon icon={CozyIcon} size="48" />} />
  </Grid>
  <Grid item>
    <SquareAppIcon name="Icon Grid" IconContent={(
      <Grid container spacing={0}>
        <Grid item xs={6}>
          <Icon icon={CozyIcon} />
        </Grid>
        <Grid item xs={6}>
          <Icon icon={CozyIcon} />
        </Grid>
          <Grid item xs={6}>
            <Icon icon={CozyIcon} />
          </Grid>
          <Grid item xs={6}>
            <Icon icon={CozyIcon} />
          </Grid>
        </Grid>
    )} />
  </Grid>
</Grid>
```
