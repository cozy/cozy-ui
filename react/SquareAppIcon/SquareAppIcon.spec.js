/* eslint-env jest */

import { render } from '@testing-library/react'
import React from 'react'

import CozyClient from 'cozy-client'

import SquareAppIcon from '.'
import Icon from '../Icon'
import CozyIcon from '../Icons/Cozy'
import DemoProvider from '../providers/DemoProvider'

const appMock = {
  slug: 'test',
  name: 'Test'
}

const client = new CozyClient({})
const Wrapper = props => {
  return (
    <DemoProvider client={client}>
      <SquareAppIcon {...props} />
    </DemoProvider>
  )
}

describe('SquareAppIcon component', () => {
  it('should render an app correctly with the given name', () => {
    const root = render(<Wrapper app={appMock} name="modified" />)
    expect(root.getByText('modified')).toBeTruthy()
    expect(root.getByTestId('square-app-icon')).toMatchSnapshot()
  })

  it('should render an app correctly with the app name', () => {
    const root = render(<Wrapper app={appMock} />)
    expect(root.getByText('Test')).toBeTruthy()
    expect(root.getByTestId('square-app-icon')).toMatchSnapshot()
  })

  it('should render an app with the app slug if no name prop and app is a string', () => {
    const root = render(<Wrapper app="testslug" />)
    expect(root.getByText('testslug')).toBeTruthy()
    expect(root.getByTestId('square-app-icon')).toMatchSnapshot()
  })

  it('should render correctly an app in maintenance state', () => {
    const root = render(<Wrapper app={appMock} variant="maintenance" />)
    expect(root.getByTestId('square-app-icon')).toMatchSnapshot()
  })

  it('should render correctly an app in ghost state', () => {
    const root = render(<Wrapper app={appMock} variant="ghost" />)
    expect(root.getByTestId('square-app-icon')).toMatchSnapshot()
  })

  it('should render correctly an app in error state', () => {
    const root = render(<Wrapper app={appMock} variant="error" />)
    expect(root.getByTestId('square-app-icon')).toMatchSnapshot()
  })

  it('should render correctly an app in add state', () => {
    const root = render(<Wrapper variant="add" />)
    expect(root.getByTestId('square-app-icon')).toMatchSnapshot()
  })

  it('should render correctly an app in shortcut state', () => {
    const root = render(<Wrapper variant="shortcut" name="shortcut" />)
    expect(root.getByTestId('square-app-icon')).toMatchSnapshot()
  })

  it('should display icon-content an app with icon content in shortcut state', () => {
    const { queryByTestId } = render(
      <Wrapper
        variant="shortcut"
        name="shortcut"
        IconContent={<Icon data-testid="icon-content" icon={CozyIcon} />}
      />
    )
    expect(queryByTestId('icon-content')).toBeTruthy()
  })

  it('should render correctly an app with custom content', () => {
    const root = render(
      <Wrapper name="custom icon" IconContent={<Icon icon={CozyIcon} />} />
    )
    expect(root.getByTestId('square-app-icon')).toMatchSnapshot()
  })
})
