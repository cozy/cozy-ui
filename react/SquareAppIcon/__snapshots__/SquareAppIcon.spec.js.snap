// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SquareAppIcon component should render an app correctly with the app name 1`] = `
<div
  class="makeStyles-tileWrapper-11"
  data-testid="square-app-icon"
>
  <span
    class="MuiBadge-root"
  >
    <div
      class="styles__c-spinner___1snK7 styles__SquareAppIcon-spinner___o0LO1 u-m-0"
    >
      <svg
        aria-busy="true"
        class="styles__icon___23x3R styles__icon--spin___ybfC1"
        height="16"
        role="progressbar"
        style="fill: var(--spinnerColor);"
        viewBox="0 0 32 32"
        width="16"
      >
        <path
          d="M16 0a16 16 0 000 32 16 16 0 000-32m0 4a12 12 0 010 24 12 12 0 010-24"
          opacity="0.25"
        />
        <path
          d="M16 0a16 16 0 0116 16h-4A12 12 0 0016 4z"
        />
      </svg>
      
    </div>
    <span
      class="MuiBadge-root styles__SquareAppIcon-wrapper___2SEuM makeStyles-shadow-9"
    >
      <div
        class="styles__SquareAppIcon-icon-container___39MRl"
      >
        <svg
          class="styles__c-app-icon___2_O40 styles__c-app-icon-default___3CEmt styles__icon___23x3R"
          height="100%"
          style="fill: var(--coolGrey);"
          viewBox="0 0 16 16"
          width="100%"
        >
          <path
            d="M1 11.009V5.5c0-.55.39-.773.872-.498l5.256 3.003c.476.272.872.944.872 1.495v5.508c0 .549-.39.772-.872.497l-5.256-3.003C1.396 12.231 1 11.56 1 11.01zm15 0c0 .55-.396 1.222-.872 1.494l-5.256 3.003c-.481.275-.872.052-.872-.497V9.5c0-.55.396-1.223.872-1.495l5.256-3.003c.481-.275.872-.052.872.498v5.508zM9.35 6.982c-.47.288-1.237.284-1.7 0l-4.8-2.954c-.47-.29-.463-.732.027-.995L7.623.477c.485-.261 1.264-.264 1.754 0l4.746 2.556c.485.26.49.71.027.995l-4.8 2.954z"
          />
        </svg>
      </div>
      <span
        class="MuiBadge-badge badgeSizeLarge MuiBadge-anchorOriginTopRightRectangular"
      />
    </span>
    <span
      class="MuiBadge-badge Component-qualifier-12 MuiBadge-anchorOriginBottomRightRectangular MuiBadge-invisible"
    />
  </span>
  <h6
    class="MuiTypography-root makeStyles-name-7 u-spacellipsis MuiTypography-h6 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
  >
    Test
  </h6>
</div>
`;

exports[`SquareAppIcon component should render an app correctly with the given name 1`] = `
<div
  class="makeStyles-tileWrapper-5"
  data-testid="square-app-icon"
>
  <span
    class="MuiBadge-root"
  >
    <div
      class="styles__c-spinner___1snK7 styles__SquareAppIcon-spinner___o0LO1 u-m-0"
    >
      <svg
        aria-busy="true"
        class="styles__icon___23x3R styles__icon--spin___ybfC1"
        height="16"
        role="progressbar"
        style="fill: var(--spinnerColor);"
        viewBox="0 0 32 32"
        width="16"
      >
        <path
          d="M16 0a16 16 0 000 32 16 16 0 000-32m0 4a12 12 0 010 24 12 12 0 010-24"
          opacity="0.25"
        />
        <path
          d="M16 0a16 16 0 0116 16h-4A12 12 0 0016 4z"
        />
      </svg>
      
    </div>
    <span
      class="MuiBadge-root styles__SquareAppIcon-wrapper___2SEuM makeStyles-shadow-3"
    >
      <div
        class="styles__SquareAppIcon-icon-container___39MRl"
      >
        <svg
          class="styles__c-app-icon___2_O40 styles__c-app-icon-default___3CEmt styles__icon___23x3R"
          height="100%"
          style="fill: var(--coolGrey);"
          viewBox="0 0 16 16"
          width="100%"
        >
          <path
            d="M1 11.009V5.5c0-.55.39-.773.872-.498l5.256 3.003c.476.272.872.944.872 1.495v5.508c0 .549-.39.772-.872.497l-5.256-3.003C1.396 12.231 1 11.56 1 11.01zm15 0c0 .55-.396 1.222-.872 1.494l-5.256 3.003c-.481.275-.872.052-.872-.497V9.5c0-.55.396-1.223.872-1.495l5.256-3.003c.481-.275.872-.052.872.498v5.508zM9.35 6.982c-.47.288-1.237.284-1.7 0l-4.8-2.954c-.47-.29-.463-.732.027-.995L7.623.477c.485-.261 1.264-.264 1.754 0l4.746 2.556c.485.26.49.71.027.995l-4.8 2.954z"
          />
        </svg>
      </div>
      <span
        class="MuiBadge-badge badgeSizeLarge MuiBadge-anchorOriginTopRightRectangular"
      />
    </span>
    <span
      class="MuiBadge-badge Component-qualifier-6 MuiBadge-anchorOriginBottomRightRectangular MuiBadge-invisible"
    />
  </span>
  <h6
    class="MuiTypography-root makeStyles-name-1 u-spacellipsis MuiTypography-h6 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
  >
    modified
  </h6>
</div>
`;

exports[`SquareAppIcon component should render an app with the app slug if no name prop and app is a string 1`] = `
<div
  class="makeStyles-tileWrapper-17"
  data-testid="square-app-icon"
>
  <span
    class="MuiBadge-root"
  >
    <div
      class="styles__c-spinner___1snK7 styles__SquareAppIcon-spinner___o0LO1 u-m-0"
    >
      <svg
        aria-busy="true"
        class="styles__icon___23x3R styles__icon--spin___ybfC1"
        height="16"
        role="progressbar"
        style="fill: var(--spinnerColor);"
        viewBox="0 0 32 32"
        width="16"
      >
        <path
          d="M16 0a16 16 0 000 32 16 16 0 000-32m0 4a12 12 0 010 24 12 12 0 010-24"
          opacity="0.25"
        />
        <path
          d="M16 0a16 16 0 0116 16h-4A12 12 0 0016 4z"
        />
      </svg>
      
    </div>
    <span
      class="MuiBadge-root styles__SquareAppIcon-wrapper___2SEuM makeStyles-shadow-15"
    >
      <div
        class="styles__SquareAppIcon-icon-container___39MRl"
      >
        <svg
          class="styles__c-app-icon___2_O40 styles__c-app-icon-default___3CEmt styles__icon___23x3R"
          height="100%"
          style="fill: var(--coolGrey);"
          viewBox="0 0 16 16"
          width="100%"
        >
          <path
            d="M1 11.009V5.5c0-.55.39-.773.872-.498l5.256 3.003c.476.272.872.944.872 1.495v5.508c0 .549-.39.772-.872.497l-5.256-3.003C1.396 12.231 1 11.56 1 11.01zm15 0c0 .55-.396 1.222-.872 1.494l-5.256 3.003c-.481.275-.872.052-.872-.497V9.5c0-.55.396-1.223.872-1.495l5.256-3.003c.481-.275.872-.052.872.498v5.508zM9.35 6.982c-.47.288-1.237.284-1.7 0l-4.8-2.954c-.47-.29-.463-.732.027-.995L7.623.477c.485-.261 1.264-.264 1.754 0l4.746 2.556c.485.26.49.71.027.995l-4.8 2.954z"
          />
        </svg>
      </div>
      <span
        class="MuiBadge-badge badgeSizeLarge MuiBadge-anchorOriginTopRightRectangular"
      />
    </span>
    <span
      class="MuiBadge-badge Component-qualifier-18 MuiBadge-anchorOriginBottomRightRectangular MuiBadge-invisible"
    />
  </span>
  <h6
    class="MuiTypography-root makeStyles-name-13 u-spacellipsis MuiTypography-h6 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
  >
    testslug
  </h6>
</div>
`;

exports[`SquareAppIcon component should render correctly an app in add state 1`] = `
<div
  class="makeStyles-tileWrapper-41"
  data-testid="square-app-icon"
>
  <span
    class="MuiBadge-root"
  >
    <span
      class="MuiBadge-root styles__SquareAppIcon-wrapper___2SEuM styles__SquareAppIcon-wrapper-fx___3rcF6"
    >
      <div
        class="styles__SquareAppIcon-icon-container___39MRl"
      >
        <svg
          class="styles__icon___23x3R"
          height="16"
          style="fill: #fff;"
          viewBox="0 0 16 16"
          width="16"
        >
          <defs>
            <path
              d="M7 0h2v7h7v2H9v7H7V9H0V7h7z"
              id="plus_svg__a"
            />
          </defs>
          <use
            fill-rule="evenodd"
            xlink:href="#plus_svg__a"
          />
        </svg>
      </div>
      <span
        class="MuiBadge-badge badgeSizeLarge MuiBadge-anchorOriginTopRightRectangular"
      />
    </span>
    <span
      class="MuiBadge-badge Component-qualifier-42 MuiBadge-anchorOriginBottomRightRectangular MuiBadge-invisible"
    />
  </span>
  <h6
    class="MuiTypography-root makeStyles-name-37 u-spacellipsis MuiTypography-h6 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
  />
</div>
`;

exports[`SquareAppIcon component should render correctly an app in error state 1`] = `
<div
  class="makeStyles-tileWrapper-35"
  data-testid="square-app-icon"
>
  <span
    class="MuiBadge-root"
  >
    <span
      class="MuiBadge-root styles__SquareAppIcon-wrapper___2SEuM makeStyles-shadow-33"
    >
      <div
        class="styles__SquareAppIcon-icon-container___39MRl"
      >
        <svg
          class="styles__c-app-icon___2_O40 styles__c-app-icon-default___3CEmt styles__icon___23x3R"
          height="100%"
          style="fill: var(--coolGrey);"
          viewBox="0 0 16 16"
          width="100%"
        >
          <path
            d="M1 11.009V5.5c0-.55.39-.773.872-.498l5.256 3.003c.476.272.872.944.872 1.495v5.508c0 .549-.39.772-.872.497l-5.256-3.003C1.396 12.231 1 11.56 1 11.01zm15 0c0 .55-.396 1.222-.872 1.494l-5.256 3.003c-.481.275-.872.052-.872-.497V9.5c0-.55.396-1.223.872-1.495l5.256-3.003c.481-.275.872-.052.872.498v5.508zM9.35 6.982c-.47.288-1.237.284-1.7 0l-4.8-2.954c-.47-.29-.463-.732.027-.995L7.623.477c.485-.261 1.264-.264 1.754 0l4.746 2.556c.485.26.49.71.027.995l-4.8 2.954z"
          />
        </svg>
      </div>
      <span
        class="MuiBadge-badge badgeSizeLarge MuiBadge-anchorOriginTopRightRectangular MuiBadge-colorError"
      >
        <svg
          class="makeStyles-errorIcon-34 styles__icon___23x3R"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M8 0C12.4183 0 16 3.58172 16 8C16 12.4183 12.4183 16 8 16C3.58172 16 0 12.4183 0 8C0 3.58172 3.58172 0 8 0ZM8 9.75C7.30964 9.75 6.75 10.3096 6.75 11C6.75 11.6904 7.30964 12.25 8 12.25C8.69036 12.25 9.25 11.6904 9.25 11C9.25 10.3096 8.69036 9.75 8 9.75ZM8 4C7.48716 4 7.06449 4.38604 7.00673 4.88338L7 5V8C7 8.55228 7.44772 9 8 9C8.51284 9 8.93551 8.61396 8.99327 8.11662L9 8V5C9 4.44772 8.55228 4 8 4Z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </span>
    <span
      class="MuiBadge-badge Component-qualifier-36 MuiBadge-anchorOriginBottomRightRectangular MuiBadge-invisible"
    />
  </span>
  <h6
    class="MuiTypography-root makeStyles-name-31 u-spacellipsis MuiTypography-h6 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
  >
    Test
  </h6>
</div>
`;

exports[`SquareAppIcon component should render correctly an app in ghost state 1`] = `
<div
  class="makeStyles-tileWrapper-29"
  data-testid="square-app-icon"
>
  <span
    class="MuiBadge-root"
  >
    <span
      class="MuiBadge-root styles__SquareAppIcon-wrapper___2SEuM styles__SquareAppIcon-wrapper-ghost___1ZALZ styles__SquareAppIcon-wrapper-fx___3rcF6"
    >
      <div
        class="styles__SquareAppIcon-icon-container___39MRl"
      >
        <svg
          class="styles__c-app-icon___2_O40 styles__c-app-icon-default___3CEmt styles__icon___23x3R"
          height="100%"
          style="fill: var(--coolGrey);"
          viewBox="0 0 16 16"
          width="100%"
        >
          <path
            d="M1 11.009V5.5c0-.55.39-.773.872-.498l5.256 3.003c.476.272.872.944.872 1.495v5.508c0 .549-.39.772-.872.497l-5.256-3.003C1.396 12.231 1 11.56 1 11.01zm15 0c0 .55-.396 1.222-.872 1.494l-5.256 3.003c-.481.275-.872.052-.872-.497V9.5c0-.55.396-1.223.872-1.495l5.256-3.003c.481-.275.872-.052.872.498v5.508zM9.35 6.982c-.47.288-1.237.284-1.7 0l-4.8-2.954c-.47-.29-.463-.732.027-.995L7.623.477c.485-.261 1.264-.264 1.754 0l4.746 2.556c.485.26.49.71.027.995l-4.8 2.954z"
          />
        </svg>
      </div>
      <span
        class="MuiBadge-badge badgeSizeLarge MuiBadge-anchorOriginTopRightRectangular"
      />
    </span>
    <span
      class="MuiBadge-badge Component-qualifier-30 MuiBadge-anchorOriginBottomRightRectangular MuiBadge-invisible"
    />
  </span>
  <h6
    class="MuiTypography-root makeStyles-name-25 u-spacellipsis MuiTypography-h6 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
  >
    Test
  </h6>
</div>
`;

exports[`SquareAppIcon component should render correctly an app in maintenance state 1`] = `
<div
  class="makeStyles-tileWrapper-23"
  data-testid="square-app-icon"
>
  <span
    class="MuiBadge-root"
  >
    <span
      class="MuiBadge-root styles__SquareAppIcon-wrapper___2SEuM styles__SquareAppIcon-wrapper-maintenance___2ne2n makeStyles-shadow-21"
    >
      <div
        class="styles__SquareAppIcon-icon-container___39MRl"
      >
        <svg
          class="styles__c-app-icon___2_O40 styles__c-app-icon-default___3CEmt styles__icon___23x3R"
          height="100%"
          style="fill: var(--coolGrey);"
          viewBox="0 0 16 16"
          width="100%"
        >
          <path
            d="M1 11.009V5.5c0-.55.39-.773.872-.498l5.256 3.003c.476.272.872.944.872 1.495v5.508c0 .549-.39.772-.872.497l-5.256-3.003C1.396 12.231 1 11.56 1 11.01zm15 0c0 .55-.396 1.222-.872 1.494l-5.256 3.003c-.481.275-.872.052-.872-.497V9.5c0-.55.396-1.223.872-1.495l5.256-3.003c.481-.275.872-.052.872.498v5.508zM9.35 6.982c-.47.288-1.237.284-1.7 0l-4.8-2.954c-.47-.29-.463-.732.027-.995L7.623.477c.485-.261 1.264-.264 1.754 0l4.746 2.556c.485.26.49.71.027.995l-4.8 2.954z"
          />
        </svg>
      </div>
      <span
        class="MuiBadge-badge badgeSizeLarge MuiBadge-anchorOriginTopRightRectangular"
      />
    </span>
    <span
      class="MuiBadge-badge Component-qualifier-24 MuiBadge-anchorOriginBottomRightRectangular MuiBadge-invisible"
    />
  </span>
  <h6
    class="MuiTypography-root makeStyles-name-19 u-spacellipsis MuiTypography-h6 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
  >
    Test
  </h6>
</div>
`;

exports[`SquareAppIcon component should render correctly an app in shortcut state 1`] = `
<div
  class="makeStyles-tileWrapper-47"
  data-testid="square-app-icon"
>
  <span
    class="MuiBadge-root"
  >
    <span
      class="MuiBadge-root styles__SquareAppIcon-wrapper___2SEuM makeStyles-shadow-45"
      style="background-color: rgb(61, 166, 126);"
    >
      <h2
        class="MuiTypography-root makeStyles-letter-44 MuiTypography-h2 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
      >
        S
      </h2>
      <span
        class="MuiBadge-badge badgeSizeLarge MuiBadge-anchorOriginTopRightRectangular"
      />
    </span>
    <span
      class="MuiBadge-badge Component-qualifier-48 MuiBadge-anchorOriginBottomRightRectangular"
    >
      <svg
        class="styles__icon___23x3R"
        height="10"
        viewBox="0 0 16 16"
        width="10"
      >
        <path
          d="M9 0v2h3.5L6 8.5 7.5 10 14 3.5V7h2V1.003A.996.996 0 0014.997 0H9zM7 2V0H1.003A1 1 0 000 1v14c0 .552.445 1 1 1h14c.552 0 1-.438 1-1.003V9h-2v5H2V2h5z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
  </span>
  <h6
    class="MuiTypography-root makeStyles-name-43 u-spacellipsis MuiTypography-h6 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
  >
    shortcut
  </h6>
</div>
`;

exports[`SquareAppIcon component should render correctly an app with custom content 1`] = `
<div
  class="makeStyles-tileWrapper-59"
  data-testid="square-app-icon"
>
  <span
    class="MuiBadge-root"
  >
    <div
      class="styles__c-spinner___1snK7 styles__SquareAppIcon-spinner___o0LO1 u-m-0"
    >
      <svg
        aria-busy="true"
        class="styles__icon___23x3R styles__icon--spin___ybfC1"
        height="16"
        role="progressbar"
        style="fill: var(--spinnerColor);"
        viewBox="0 0 32 32"
        width="16"
      >
        <path
          d="M16 0a16 16 0 000 32 16 16 0 000-32m0 4a12 12 0 010 24 12 12 0 010-24"
          opacity="0.25"
        />
        <path
          d="M16 0a16 16 0 0116 16h-4A12 12 0 0016 4z"
        />
      </svg>
      
    </div>
    <span
      class="MuiBadge-root styles__SquareAppIcon-wrapper___2SEuM makeStyles-shadow-57"
    >
      <div
        class="styles__SquareAppIcon-icon-container___39MRl"
      >
        <svg
          class="styles__icon___23x3R"
          height="16"
          viewBox="0 0 52 52"
          width="16"
        >
          <path
            d="M38.231 44H13.769C6.175 44 0 37.756 0 30.08c0-3.66 1.394-7.117 3.927-9.733 2.219-2.29 5.093-3.715 8.203-4.086a13.887 13.887 0 014.042-8.292A13.608 13.608 0 0125.801 4c3.62 0 7.04 1.407 9.629 3.968a13.897 13.897 0 014.038 8.25C46.482 16.853 52 22.828 52 30.082 52 37.756 45.82 44 38.23 44h.001zm-.163-3.001h.104c5.97 0 10.828-4.91 10.828-10.947 0-6.035-4.857-10.946-10.828-10.946h-.11c-.779 0-1.417-.627-1.435-1.417C36.492 11.794 31.637 7 25.803 7c-5.835 0-10.692 4.796-10.826 10.69a1.445 1.445 0 01-1.403 1.42C7.744 19.244 3 24.153 3 30.052 3 36.09 7.857 41 13.828 41h.088l.035-.002c.03 0 .062 0 .093.002h24.021l.003-.001zm-4.302-11.776c-.875-.585-.918-1.659-.92-1.706A.52.52 0 0032.32 27a.523.523 0 00-.506.536c.002.039.016.543.251 1.137a7.99 7.99 0 01-11.138.019 3.554 3.554 0 00.257-1.155.523.523 0 00-.503-.536.526.526 0 00-.528.515c0 .043-.042 1.121-.92 1.706a.536.536 0 00-.15.731.51.51 0 00.714.154c.225-.15.414-.322.572-.505a9.006 9.006 0 0012.251-.01c.16.184.35.36.582.515a.503.503 0 00.281.085.516.516 0 00.432-.24.537.537 0 00-.15-.731v.002z"
            fill="#297EF2"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <span
        class="MuiBadge-badge badgeSizeLarge MuiBadge-anchorOriginTopRightRectangular"
      />
    </span>
    <span
      class="MuiBadge-badge Component-qualifier-60 MuiBadge-anchorOriginBottomRightRectangular MuiBadge-invisible"
    />
  </span>
  <h6
    class="MuiTypography-root makeStyles-name-55 u-spacellipsis MuiTypography-h6 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
  >
    custom icon
  </h6>
</div>
`;
